syntheseus/search/chem.py

```diff
@@ -15,8 +15,10 @@ class ReactionMetaData(TypedDict, total=False):
     cost: float
     template: str
     source: str  # any explanation of the source of this reaction
+    probability: float  # probability used for multi-step search
```

Comment:
I just realized it was also used in a previous PR. Still, @kmaziarz or @fiberleif do you have any opinion about whether "score" or "probability" is a better name?

Reply:
Yes, l added a `probability` attribute here since l found this attribute will be used in some other places we're using (e.g., ReactionModelBasedEvaluator, etc)

Reply:
Fair, but I think `probability` is an loaded term with a lot of connotations (e.g. normalization) which the values of `probability` may not satisfy. This is why I originally used the word `score`.

Reply:
Understood! Perhaps for this PR, we can continue to use the term "probability" to maintain compatibility with previous PRs. However, if we ultimately determine that "score" is more appropriate, we can create a new PR to fully implement the change.

Reply:
Yeah, as you both notice, `probability` is already used to store single-step model probability in `SyntheseusBackwardReactionModel`. We could switch to `score` in a follow-up PR if we feel this is better, although I did check in the past that for all single-step models what is placed under `probability` is actually in `[0, 1]`, so it seems reasonably consistent with the definition of "probability"...